dbms/src/IO/IOThreadPool.h

```diff
@@ -25,6 +26,8 @@ template <typename Type>
 class IOThreadPool
 {
     friend void adjustThreadPoolSize(const Settings & settings, size_t logical_cores);
+    friend void setLoadMetadataThreadPool(const Context & context);
```

Comment:
It is a bad idea to introduce business logic to a common utility. Please consider to remove it.